src/org/labkey/targetedms/outliers/OutlierGenerator.java

```diff
@@ -64,31 +65,21 @@ private String getEachSeriesTypePlotDataSql(int seriesIndex, int id, String sche
             StringBuilder filterClause = new StringBuilder("SampleFileId.ReplicateId in (");
             var intersect = "";
             var selectSql = "(SELECT ReplicateId FROM targetedms.ReplicateAnnotation WHERE ";
-            var escapeQuote = "\'";
             for (AnnotationGroup annotation : annotationGroups)
             {
                 filterClause.append(intersect)
                         .append(selectSql)
-                        .append(" Name=")
-                        .append(escapeQuote)
-                        .append(annotation.getName())
-                        .append(escapeQuote);
+                        .append(" Name='")
+                        .append(annotation.getName().replace("'", "''"))
+                        .append("'");
+
 
                 var annotationValues = annotation.getValues();
                 if (!annotationValues.isEmpty())
                 {
-                    filterClause.append(" AND  Value IN (")
-                            .append(escapeQuote)
-                            .append(annotationValues.get(0))
-                            .append(escapeQuote);
-
-                    for (int i = 1; i < annotationValues.size(); i++)
-                    {
-                        filterClause.append(",")
-                                .append(escapeQuote)
-                                .append(annotationValues.get(i))
-                                .append(escapeQuote);
-                    }
+                    var quoteEscapedVals = annotationValues.stream().map(s -> s.replace("'", "''")).collect(Collectors.toList());
+                    var vals = "'" + StringUtils.join(quoteEscapedVals, "','") + "'";
+                    filterClause.append(" AND  Value IN (").append(vals);
```

Comment:
I haven't tried running this, but the parentheses seems suspicious. Shouldn't there be a closing paren inside of the if() statement's body?